packages/cli/src/utils/errors.ts

```diff
@@ -1,54 +1,47 @@
-// Based on hardhat's errors and errors-list (MIT)
-// https://github.com/NomicFoundation/hardhat/tree/main/packages/hardhat-core
-
-export interface ErrorDescriptor {
-  message: (...args: string[]) => string;
-  description: string;
+import chalk from "chalk";
+import { ZodError } from "zod";
+import { fromZodError, ValidationError } from "zod-validation-error";
+
+// Wrapper with preset styles, only requires a `prefix`
+export function fromZodErrorCustom(error: ZodError, prefix: string) {
+  return fromZodError(error, {
+    prefix: chalk.red(prefix),
+    prefixSeparator: "\n- ",
+    issueSeparator: "\n- ",
+  });
 }
 
-export class CustomError extends Error {
-  constructor(message: string, public readonly parent?: Error) {
-    super(message);
-
-    this.name = this.constructor.name;
-  }
+export class NotInsideProjectError extends Error {
+  name = "NotInsideProjectError";
+  message = "You are not inside a MUD project";
 }
```

Comment:
small thing: I would probably add

```ts
export class MudCliError extends Error {}
```
then
```ts
export class NotInsideProjectError extends MudCliError {
  ...
}
```

just so downstream folks can easily catch all CLI errors in one conditional

(I am thinking we might eventually have a centralized `MudError` but a cli-specific one in the meantime is still handy)

Reply:
That said, I'm not sure how these interact with Zod error classes.

Reply:
> just so downstream folks can easily catch all CLI errors in one conditional

There aren't really supposed to be downstream folks rn, the renderer is cli-only currently, and everything is handled by `logError`.
When I do make it standalone that sounds like a good thing to add though. Zod errors can probably be wrapped to also be identifiable somehow